src/test/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParserTest.java

```diff
@@ -831,6 +831,42 @@ public void testOutOfRangesArguments(final String[] argv) throws Exception {
         clp.parseArguments(System.err, argv);
     }
 
+    @Test(expectedExceptions = CommandLineException.CommandLineParserInternalException.class)
+    public void testHiddenRequiredArgumentThrowException() throws Exception {
+        @CommandLineProgramProperties(summary = "tool with required and hidden argument",
+                oneLineSummary = "tool with required and hidden argument",
+                programGroup = TestProgramGroup.class)
+        final class ToolWithRequiredHiddenArgument {
+            @Argument(fullName = "hidden", shortName = "hidden", doc = "Hidden argument", optional = false)
+            @Hidden
+            public Integer hidden;
+        }
+        new CommandLineArgumentParser(new ToolWithRequiredHiddenArgument());
+    }
+
+    @Test
+    public void testHiddenArguments() throws Exception {
+        @CommandLineProgramProperties(summary = "test",
+                oneLineSummary = "test",
+                programGroup = TestProgramGroup.class)
+        final class ToolWithHiddenArgument {
+            @Argument(fullName = "hidden", shortName = "hidden", doc = "Hidden argument", optional = true)
```

Comment:
Can you change the fullName of this argument to something more unique and less likely to show up in the usage output by accident, like "hiddenTestArgument".